dotnet-interactive.Tests/HttpApiTests.cs

```diff
@@ -246,6 +248,72 @@ public async Task unimplemented_lsp_handlers_return_empty_string(Language langua
             responseJson.Should().BeEmpty();
         }
 
+        [Fact]
+        public async Task lsp_methods_run_deferred_commands()
+        {
+            // declare a variable in deferred code
+            var kernelBase = GetServer().Kernel as KernelBase;
+            kernelBase.DeferCommand(new SubmitCode("var one = 1;"));
+
+            // it's not defined
+            var response = await GetServer().HttpClient.GetAsync($"/variables/{Language.CSharp.LanguageName()}/one");
+            response.StatusCode.Should().Be(HttpStatusCode.NotFound);
+
+            // ensure we can get http info about it
+            var encoded = Convert.ToBase64String(Encoding.UTF8.GetBytes("Console.WriteLine(one);"));
```

Comment:
This trivia shouldn't be in a test. Did we encapsulate this into a class somewhere?

Reply:
No, because this is always handled by the client (`lsp.js`), but we have to fake that here because we don't have a client.